### PR TITLE
show discontinued in pub outdated

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: lib/src/command/outdated.dart [hunk 1]
```diff
@@ -427,14 +432,22 @@ Future<void> _outputHuman(
   required String directory,
 }) async {
   final directoryDesc = directory == '.' ? '' : ' in $directory';
-  log.message(mode.explanation(directoryDesc) + '\n');
+
+  log.message(mode.explanation(directoryDesc));
+  if (rows.firstWhereOrNull((package) => package.isDiscontinued) != null) {
```
Comment (sigurdm) [inline lib/src/command/outdated.dart:437]: Could use 
```suggestion
  if (rows.any((package) => package.isDiscontinued)) {
```
```suggestion
  if (rows.firstWhereOrNull((package) => package.isDiscontinued) != null) {
```

Comment (sigurdm) [inline lib/src/command/outdated.dart:437]: Could use 
```suggestion
  if (rows.any((package) => package.isDiscontinued)) {
```
```suggestion
  if (rows.firstWhereOrNull((package) => package.isDiscontinued) != null) {
```

Comment (szakarias) [reply lib/src/command/outdated.dart]: Removed, due to layout change.

**Comment by szakarias** [comment]
PTAL

File: test/testdata/goldens/outdated/outdated_test/show discontinued.txt [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+direct dependencies:
+bar           1.0.0    1.0.0       1.0.0       1.0.0                 
+foo           1.2.3    1.2.3       1.2.3       1.2.3 (discontinued)  
+
+transitive dependencies:
+transitive    1.2.3    1.2.3       1.2.3       1.2.3                 
+You are already using the newest resolvable versions listed in the 'Resolvable' column.
+Newer versions, listed in 'Latest', may not be mutually compatible.
+foo
```
Comment (sigurdm) [inline test/testdata/goldens/outdated/outdated_test/show discontinued.txt:44]: This somehow looks off when there is so little information.

At the very least we need a newline before this section

File: test/testdata/goldens/outdated/outdated_test/show discontinued.txt [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    discontinued
```
Comment (sigurdm) [inline test/testdata/goldens/outdated/outdated_test/show discontinued.txt:45]: We need a full stop `.` after this line

File: test/testdata/goldens/outdated/outdated_test/show discontinued.txt (hunk 2 above)
Comment (szakarias) [reply test/testdata/goldens/outdated/outdated_test/show discontinued.txt:44]: When shown in the terminal, the name of the package is in bold. We could also add a newline.

File: test/testdata/goldens/outdated/outdated_test/show discontinued.txt (hunk 3 above)
Comment (szakarias) [reply test/testdata/goldens/outdated/outdated_test/show discontinued.txt:45]: done.

File: test/testdata/goldens/outdated/outdated_test/show discontinued.txt [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+
+Package Name  Current  Upgradable  Resolvable  Latest  
+
+direct dependencies:
+bar           ✗1.0.0   ✗1.0.0      ✗1.0.0      ✗1.0.0  
+foo           ✗1.2.3   ✗1.2.3      ✗1.2.3      ✗1.2.3  
+You are already using the newest resolvable versions listed in the 'Resolvable' column.
+Newer versions, listed in 'Latest', may not be mutually compatible.
+foo
+    discontinued
```
Comment (jonasfj) [inline test/testdata/goldens/outdated/outdated_test/show discontinued.txt:129]: Shouldn't we have a few more empty lines?

```
You are already using the newest resolvable versions listed in the 'Resolvable' column.
Newer versions, listed in 'Latest', may not be mutually compatible.

foo:
    Package foo has been discontinued, (replaced by ...)
```

Or maybe we should do something like:

```
You are already using the newest resolvable versions listed in the 'Resolvable' column.
Newer versions, listed in 'Latest', may not be mutually compatible.

Notes for individual packages:

foo:
    Package foo has been discontinued, (replaced by ...)
```

Comment (szakarias) [reply test/testdata/goldens/outdated/outdated_test/show discontinued.txt:129]: Done.

File: test/testdata/goldens/outdated/outdated_test/show discontinued.txt [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+$ pub outdated --no-color --mode=null-safety
+Showing dependencies that are currently not opted in to null-safety.
+[✗] indicates versions without null safety support.
+[✓] indicates versions opting in to null safety.
+
+Package Name  Current  Upgradable  Resolvable  Latest  
+
+direct dependencies:
+bar           ✗1.0.0   ✗1.0.0      ✗1.0.0      ✗1.0.0  
+baz           ✗1.0.0   ✗1.0.0      ✗1.0.0      ✗1.0.0
```
Comment (sigurdm) [inline test/testdata/goldens/outdated/outdated_test/show discontinued.txt:90]: The last column should also read "outdated" in null-safety mode

**Review by sigurdm — APPROVED** [review]
LGTM with a single nit

File: test/testdata/goldens/outdated/outdated_test/show discontinued.txt (hunk 5 above)
Comment (szakarias) [reply test/testdata/goldens/outdated/outdated_test/show discontinued.txt]: done

**Review by sigurdm — APPROVED** [review]

**Pull request merged**